enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/server/CoreServerModule.java

```diff
@@ -192,11 +194,12 @@ public CoreServerModule( IdentityModule identityModule, final PlatformModule pla
         ApplicationProtocolRepository catchupProtocolRepository = new ApplicationProtocolRepository( ApplicationProtocols.values(), supportedCatchupProtocols );
         ModifierProtocolRepository modifierProtocolRepository = new ModifierProtocolRepository( ModifierProtocols.values(), supportedModifierProtocols );
 
+        ExecutorLifecycleAdapter checkpointExecutorService = life.add( new ExecutorLifecycleAdapter( Executors::newSingleThreadExecutor ) );
```

Comment:
I think its recommended to use JobScheduler and not create Executors. Perhaps CentralJobScheduler can be used?

Reply:
And the JobScheduler instance is readily available in a field on the PlatformModule.

Reply:
Ah, wasn't aware, thanks for finding that!